### ISSUE TITLE
Add RabbitMQ install roles

### ISSUE THREAD

**Issue by cognifloyd** [ansible-collections/community.rabbitmq#45]
##### SUMMARY
Add generic roles for installing rabbitmq. They should span simple use-cases (single server) to clustered server.
Since this is a collection, we can break the installation down into multiple roles so that users can pick and choose the roles they need.

I'm thinking of something similar to the roles in [`community.mongodb`](https://github.com/ansible-collections/community.rabbitmq).

Are there any suggestions on how to break down the set of install roles?
Once there is some feedback, I can work on either porting or re-writing some external roles.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
Misc rabbitmq roles

**Comment by cognifloyd** [comment]
One of my goals in putting these generic roles together here, is so that disparate communities can work together on installation roles for common dependencies.

I've heard that the `openstack.rabbitmq_server` role is a good starting point. So, I'll start looking through that to see how many different roles we might want here.

**Comment by odyssey4me** [comment]
I'm open to this idea. I'm also open to seeing thoughts on what the roles may be. The [openstack-ansible role](https://opendev.org/openstack/openstack-ansible-rabbitmq_server) is somewhat monolithic which has its benefits and drawbacks. It might be worthwhile to break things out into multiple roles - for example, one to do initial install, one to configure and perhaps one specially suited to cluster management. Having multiple roles may make each one simpler to maintain... That said, I've not thought it through in much detail.

**Comment by cognifloyd** [comment]
Yes. I want to split this out into multiple roles. We could have one meta role that orchestrates the other roles to ease transition from openstack's role.

I've been busy with a new job, so I'll get back to this very soon. :)

**Comment by cognifloyd** [comment]
So here are the roles I think we could have:
- `community.rabbitmq.rabbitmq_repository`: Add the deb/rpm/etc package repository
- `community.rabbitmq.rabbitmq_install`: Install rabbitmq itself
- `community.rabbitmq.rabbitmq_config`: Setup rabbitmq.conf and optionally add plugins, policies, vhosts, users

We might end up with another utility role or two, but that's the first stab at splitting it up.

**Comment by Andersson007** [comment]
@odyssey4me @aitorpazos what do you think of the suggested roles? ^

**Comment by Andersson007** [comment]
The `battery included` approach sounds good to me:) I'm not sure about the particular modules, that's why we're asking you, folks.

**Comment by DanceMore** [comment]
$newJob "likes ansible" but they've never bothered to use "modules" or "galaxy" or advanced features like that....

I'm trying to modernize things and I'm pretty shocked to see that `community.rabbitmq` can't configure repos and install the packages for me ...?
